openstack_exporter/exporter.py

```diff
@@ -0,0 +1,88 @@
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import click
+import logging
+import os
+import sys
+import time
+
+from prometheus_client.core import REGISTRY
+from prometheus_client import start_http_server
+import yaml
+
+from openstack_exporter.collectors import cinderbackend
+
+
+def run_prometheus_server(port, openstack_config):
+    start_http_server(int(port))
+    REGISTRY.register(cinderbackend.CinderBackendCollector(openstack_config))
+    while True:
+        time.sleep(1)
+
+
+def get_config(config_file):
+    if os.path.exists(config_file):
+        try:
+            with open(config_file) as f:
+                config = yaml.load(f, Loader=yaml.FullLoader)
+        except IOError as e:
+            logging.error("Couldn't open configuration file: " + str(e))
+        return config
+    else:
+        logging.error("Config file doesn't exist: " + config_file)
+        exit(0)
+
+
+@click.command()
+@click.option("--port", metavar="<port>", default=9102,
+              help="specify exporter serving port")
+@click.option("-d", "--debug", metavar="<debug>", is_flag=True,
+              default=False, help="enable debug")
```

Comment:
Debug does not switch anything. Handled by config file

Reply:
removed